xarray/core/merge.py

```diff
@@ -700,6 +703,9 @@ def merge(
         - "override": skip comparing and copy attrs from the first dataset to
           the result.
 
+        If a callable, it must expect a sequence of ``attrs`` dicts as its only
+        parameter.
```

Comment:
Let's also make it accept a second argument "context"

Reply:
done, although `context` will always be `None` until I can figure out if it's possible to avoid manually creating the context object.